src/utils/components/DiskModal/utils/helpers.ts

```diff
@@ -155,12 +155,12 @@ export const getDataVolumeFromState = ({
   const dvName =
     resultVolume?.dataVolume?.name ||
     resultVolume?.persistentVolumeClaim?.claimName ||
-    `${vm?.metadata?.name}-${diskState.diskName}`;
+    nameWithoutParameter(
```

Comment:
do we need it at all?

Reply:
Yes. It's part of a bug fix: https://issues.redhat.com/browse/CNV-28231.

Reply:
i removed nameWithoutParameter in another pr : https://github.com/kubevirt-ui/kubevirt-plugin/pull/1977

Reply:
@upalatucci, I remember you removing it, as you couldn't find the reason why it was added in the first place. Now that it's removed, do we have a regression with the bug link @pcbailey mentioned?

Reply:
When the user tries to add a new disk on a template, we need to keep the parameter in the name.

Without a parameter ${NAME}, we can use the template to create different vms, but all those vms will use the same shared disk. This is not the result that the user intend to have when creates a disk in the template. 

I suggest to create a PVC from the uploaded file with a random string without any parameters (uploadData is the method responsible for that) and use it as a source for the datavolume. The data volume has to have the parameter in its name. With this mechanism we'll clone the same PVC to each VM created with that template.